src/nbl/video/COpenGLCommandBuffer.cpp

```diff
@@ -96,6 +101,18 @@ namespace video
         }
     }
 
+    bool COpenGLCommandBuffer::reset(uint32_t _flags)
+    {
+        if (!(m_cmdpool->getCreationFlags() & IGPUCommandPool::ECF_RESET_COMMAND_BUFFER_BIT))
+            return false;
+
+        freeSpaceInCmdPool();
+        m_commands.clear();
```

Comment:
btw why not implement the commandbuffer as a forward linked list, the commands are allocated from pools already, you can get rid of a dynamic allocation `vector` by simply storing the pointer/link to the next command in the previous command